### PR TITLE
macOS fixes

### PR DESCRIPTION
**Pull request by rcombs**
- Update constant names to fix some deprecation warnings
- Fix non-ASCII CoreAudio device descriptions not being returned

### CODE REVIEW

**Comment by Akemi** [comment]
we had the first commit (`Cocoa: fix deprecation warnings`) before on #6612 and it's still not acceptable like that. it fixes the deprecation warning but breaks the d'n'd functionality. one needs to do the same as https://github.com/mpv-player/mpv/commit/a8c2e2986838dccbcc4bd218b501f0bf86b36e2c#diff-73aafea4a74c6dcc7b2768925e8747cf

it also throws these warnings when dropping a file in the window.
```
2020-02-26 19:24:09.172 mpv[6159:315806] -[__NSCFString count]: unrecognized selector sent to instance 0x60700034fd40
2020-02-26 19:24:09.173 mpv[6159:315806] *** Canceling drag because exception 'NSInvalidArgumentException' (reason '-[__NSCFString count]: unrecognized selector sent to instance 0x60700034fd40') was raised during a dragging session
2020-02-26 19:24:14.030 mpv[6159:315806] -[__NSCFString count]: unrecognized selector sent to instance 0x6070005c7320
2020-02-26 19:24:14.030 mpv[6159:315806] *** Canceling drag because exception 'NSInvalidArgumentException' (reason '-[__NSCFString count]: unrecognized selector sent to instance 0x6070005c7320') was raised during a dragging session
```

the other commit seems fine.

**Comment by rcombs** [comment]
This oughtta be fine now, assuming the URL bits actually do work pre-10.13 as is implied by the Swift code.

**Comment by Akemi** [comment]
that fails on our 10.12 CI run https://travis-ci.org/github/mpv-player/mpv/jobs/673293811
i believe the `if (@available(macOS 10.13, *))` check is only available on LLVM5.0, which is xcode 9.3 minimum and that is only available on 10.13 upwards.

on a side note the 10.12 CI run was marked as optional to save some time on unrelated PRs, since it can take quite some time. for macOS related changes one has to look at the details for that.

**Comment by rcombs** [comment]
Erm, sorry, lemme fix the `@available` check…

**Comment by rcombs** [comment]
Alright, fixed. I don't think `@available` was actually an issue here; it was that `NSPasteboardType` and the related constants weren't in the headers until the 10.13 SDK (and I didn't realize we supported building with SDKs that old). Passes CI now with a compile-time SDK-version check.

**Comment by Akemi** [comment]
now i am getting.

```
2020-05-03 12:51:11.254 mpv[3282:86987] -[NSURL hasPrefix:]: unrecognized selector sent to instance 0x608000091420
2020-05-03 12:51:11.254 mpv[3282:86987] *** Canceling drag because exception 'NSInvalidArgumentException' (reason '-[NSURL hasPrefix:]: unrecognized selector sent to instance 0x608000091420') was raised during a dragging session
2020-05-03 12:51:34.436 mpv[3282:86987] -[NSURL hasPrefix:]: unrecognized selector sent to instance 0x6080000d4ba0
2020-05-03 12:51:34.436 mpv[3282:86987] *** Canceling drag because exception 'NSInvalidArgumentException' (reason '-[NSURL hasPrefix:]: unrecognized selector sent to instance 0x6080000d4ba0') was raised during a dragging session
```

won't the usage of `AvailabilityMacros.h` interfere with what we did [here](https://github.com/mpv-player/mpv/blob/master/osdep/macosx_versions.h). not in this case but in possible future ones?

**Comment by rcombs** [comment]
Realized that when I attempted to test this code, it wasn't actually running because the Swift version was being used instead. Not sure how to avoid that on modern machines, so here's a stab at a fix I can't validate myself (it passes the static analyzer, at least).
Re: AvailabilityMacros, I don't have any better ideas for detecting SDK versions; if you know anything better lmk. Otherwise it seems like we could just define our own macro names to either a hardcoded value or the SDK's depending on if the SDK's is defined or not.

**Comment by Akemi** [comment]
i just test it by setting `--vo=gpu`, that will use the old cocoa backend.

for swift i added a [macOS sdk check](https://github.com/mpv-player/mpv/commit/850b3037323ad68012c1d0b59ec00f56a9d8c09b#diff-038007df544118b88c4edbf4c79a6fa0) to our build system. it can be used like [this](https://github.com/mpv-player/mpv/commit/c540ac8485d132285071aa43a493c3829f4f0d5c). you add a check to our `wscript` if it's not available yet, so yeah we need another one for 10.13. it will add a macro `HAVE_MACOS_10_13_FEATURES` that you can use for the compile time check.

**Comment by rcombs** [comment]
Alright, this seems to be working with the test you suggested now.

File: video/out/cocoa/events_view.m [hunk 1]
```diff
@@ -35,12 +35,40 @@ @implementation MpvEventsView
...
+        return (NSPasteboardType)kUTTypeFileURL;
+    }
+}
+
+static NSPasteboardType urlType(void)
+{
+#if HAVE_MACOS_10_13_FEATURES
+    if (@available(macOS 10.13, *)) {
+        return NSPasteboardTypeURL;
+    } else
```
Comment (Akemi) [inline video/out/cocoa/events_view.m:59]: the else case here seems redundant, `(NSPasteboardType)kUTTypeURL` can be returned directly. same for the function above.

**Comment by Akemi** [comment]
i think you can squash `Cocoa: fix pasteboard deprecation warnings` and `build: add conditional compilation for macOS 10.13 SDK`. the first letter on the first commit should also be lower case.

**Pull request closed without merging**